extensions/wikia/ContributionAppreciation/ContributionAppreciationController.class.php

```diff
@@ -169,18 +169,23 @@ private function sendMail( $revisionId ) {
 		$revision = Revision::newFromId( $revisionId );
 
 		if ( $revision ) {
-			$editedPageTitle = $revision->getTitle();
-			$params = [
-				'buttonLink' => SpecialPage::getTitleFor( 'Community' )->getFullURL(),
-				'targetUser' => $revision->getUserText(),
-				'editedPageTitleText' => $editedPageTitle->getText(),
-				'editedWikiName' => $wgSitename,
-				'revisionUrl' => $editedPageTitle->getFullURL( [
-					'diff' => $revision->getId()
-				] )
-			];
-
-			$this->app->sendRequest( 'Email\Controller\ContributionAppreciationMessageController', 'handle', $params );
+			$diffAuthor = \User::newFromId( $revision->getUser() );
+
+			// we want to send appreciation email for en users only
+			if ( $diffAuthor && $diffAuthor->getGlobalPreference( 'language' ) == 'en' ) {
```

Comment:
Please extract the language-specific part of condition from here and https://github.com/Wikia/app/pull/11083/files#diff-dd7892d407f19b1887cafc528350ea56L110 to a single place, i.e.:
```
isSuportedAppreciationLang($lang) 
```